error when building

Any idea? This is under R version 3.4.4, Rcpp version 0.12.16, using gcc 6.3.0 on RHEL 6.8

```
file.cc: In function 'Rcpp::List stat_(Rcpp::CharacterVector)':
file.cc:199:24: error: expected ')' before 'PRIu64'
       sprintf(buf, "%" PRIu64, st.st_uid);
                        ^~~~~~
file.cc:212:24: error: expected ')' before 'PRIu64'
       sprintf(buf, "%" PRIu64, st.st_gid);
                        ^~~~~~
make: *** [file.o] Error 1
ERROR: compilation failed for package 'fs'
```

--- a/src/file.cc
+++ b/src/file.cc
@@ -1,3 +1,7 @@
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS 1
+#endif
+
 #include "getmode.h"
 #include "uv.h"
 
@@ -9,10 +13,6 @@
 #ifndef __WIN32
 #include <grp.h>
 #include <pwd.h>
-#endif
-
-#ifndef __STDC_FORMAT_MACROS
-#define __STDC_FORMAT_MACROS 1
 #endif
 
 #include <inttypes.h>

